### PR TITLE
Use ESLint Shared Configuration

### LINKED ISSUE #127: Use templates for variable usage within strings
Start using templates for variable usage within strings, see [here](https://github.com/sugarcrm/thorn/pull/113#discussion_r99701779).

### PR DESCRIPTION
**Pull request by rhoggSugarcrm**
This is identical to #126, but migrating it to my new fork.

Closes #127.

### CODE REVIEW

**Comment by alias-mac** [comment]
If we can make a repository already for our JS standards, it would be even better.

File: package.json [hunk 1]
```diff
@@ -26,6 +26,7 @@
     "co-mocha": "^1.2.0",
     "commander": "^2.9.0",
     "coveralls": "^2.11.16",
+    "@sugarcrm/eslint-config-sugarcrm": "git+https://github.com/sugarcrm/javascript",
```
Comment (alias-mac) [inline package.json]: I think scoped packages are normally added in the top

Comment (alias-mac) [inline package.json]: if we can publish the package today, would be great to just add this one from npm directly

File: index.js [hunk 2]
```diff
@@ -165,20 +165,18 @@ let Fixtures = {
...
+        if (!_.isArray(models)) {
             models = [models];
         }
 
         // reset `_sessionAttempt`
         this._sessionAttempt = 0;
 
         let url = utils.constructUrl(VERSION, 'bulk');
-        let params = {headers: this._headers};
+        let params = { headers: this._headers };
```
Comment (f1vlad) [inline index.js]: Despite this being ok with jslint, I think there shouldn't be white space before `headers`.

Comment (f1vlad) [inline index.js]: Despite this being ok with jslint, I think there shouldn't be white space before `headers`.

File: gulpfile.babel.js [hunk 3]
```diff
@@ -1,51 +1,51 @@
...
         .option('--ci', 'Enable CI specific options')
         .option('--path <path>', 'Set base output path')
         .parse(process.argv);
 
-    var path = commander.path || process.env.WORKSPACE || os.tmpdir();
+    let path = commander.path || process.env.WORKSPACE || os.tmpdir();
 
-    var options = {
+    let options = {
         timeout: '5000',
```
Comment (f1vlad) [inline gulpfile.babel.js]: I don't think there should be comma here.

File: index.js (hunk 2 above)
Comment (rhoggSugarcrm) [inline index.js]: I'll be removing the rule that requires this from the sugarcrm/javascript repo later tonight per what you said and Joao's [review summary from before](https://github.com/sugarcrm/thorn/pull/138#pullrequestreview-23100226).

File: tests/metadata-handler.js [hunk 4]
```diff
@@ -32,16 +36,16 @@ describe('Metadata Handler', () => {
...
+                    expect(() => Meta.generateFieldValue({ type })).to.throw(msg);
                 });
             });
         });
 
         describe('unrecognized types', () => {
             it('should throw an error', () => {
                 let msg = 'Field type i_am_not_a_real_type is not recognized.';
-                expect(() => Meta.generateFieldValue({type: 'i_am_not_a_real_type'})).to.throw(msg);
+                expect(() => Meta.generateFieldValue({ type: 'i_am_not_a_real_type' })).to.throw(msg);
```
Comment (f1vlad) [inline tests/metadata-handler.js]: Similar thing here like with `headers` and white space. White space like this doesn't look good to me when I am scanning the code.

File: index.js (hunk 2 above)
Comment (rhoggSugarcrm) [inline index.js]: Opened https://github.com/sugarcrm/javascript/pull/2

**Pull request merged**